tests/test_api.py

```diff
@@ -501,18 +531,61 @@ def test_pf_pickle_pf(self, config):
         # the rel_error is often higher than 0.5 but the tests are random
 
 
-def test_history():
-    """Test that history_'s keys are strings and values are lists.
-    """
-    data = load_boston()
-    X, y = data.data[:100], data.target[:100]
-    estimator = KerasRegressor(model=dynamic_regressor, model__hidden_layer_sizes=[])
+def force_compile_shorthand(hidden_layer_sizes, meta, compile_kwargs, params):
+    model = dynamic_regressor(
+        hidden_layer_sizes=hidden_layer_sizes, meta=meta, compile_kwargs=compile_kwargs
+    )
+    model.compile(
+        optimizer=compile_kwargs["optimizer"],
+        loss=compile_kwargs["loss"],
+        metrics=params["metrics"],
+    )
+    return model
+
+
+class TestHistory:
+    def test_history(self):
+        """Test that history_'s keys are strings and values are lists.
+        """
+        data = load_boston()
+        X, y = data.data[:100], data.target[:100]
+        estimator = KerasRegressor(
+            model=dynamic_regressor, model__hidden_layer_sizes=[]
+        )
+
+        estimator.partial_fit(X, y)
 
-    estimator.partial_fit(X, y)
+        assert isinstance(estimator.history_, dict)
+        assert all(isinstance(k, str) for k in estimator.history_.keys())
+        assert all(isinstance(v, list) for v in estimator.history_.values())
 
-    assert isinstance(estimator.history_, dict)
-    assert all(isinstance(k, str) for k in estimator.history_.keys())
-    assert all(isinstance(v, list) for v in estimator.history_.values())
+    def test_partial_fit_shorthand_metric_name(self):
+        """Test that metrics get stored in the `history_` attribute
+        by their long name (and not shorthand) and that they
+        survive a pickle round trip.
+        """
+        estimator = KerasRegressor(
+            model=force_compile_shorthand,
+            run_eagerly=True,
+            loss=KerasRegressor.r_squared,
+            model__hidden_layer_sizes=(100,),
+            metrics=["mae"],  # shorthand
+        )
+        X, y = load_boston(return_X_y=True)
+        X = X[:100]
+        y = y[:100]
+        estimator.fit(X, y)
+        assert estimator.model_.metrics_names[1] == "mae"
+        assert "mean_absolute_error" in estimator.history_
+        assert "mae" not in estimator.history_
```

Comment:
Style nit and suggested change:

``` diff
- estimator = KerasRegressor(...)
- assert "mean_absolute_error" in estimator.history_
- assert "mae" not in estimator.history_
+ est = KerasRegressor(...)
+ assert "mae" not in est.history_ and "mean_absolute_error" in est.history_
```

Reply:
Changed as suggested. Also split up the test.